src/v/config/property.h

```diff
@@ -956,5 +965,99 @@ class hidden_when_default_property : public property<T> {
         return this->value() == this->default_value();
     }
 };
+namespace detail {
+
+template<typename P>
+concept Property = requires() {
+    std::derived_from<P, base_property>;
+    typename P::value_type;
+};
+
+template<typename T>
+concept Array = detail::is_array<T>() && !reflection::is_std_optional<T>;
+
+} // namespace detail
+
+template<detail::Property P>
+class enterprise : public P {
```

Comment:
Consider adding a comment about the expected parameters and the logic included in the wrapper